design/ceph/osd-migration.md

```diff
@@ -36,139 +25,107 @@ considered in the future:
   and application data on slower media.
 - Setups with multiple OSDs per drive, though with recent Ceph releases the
   motivation for deploying this way is mostly obviated.
+- Mix of encrypted and unencrypted OSDs in the cluster.
 
 ## Proposal
+- Since migration requires destroying of the OSD and cleaning data from the disk,
+  a user confirmation in the CR is required before proceeding.
+- Operator would identify the OSDs that need migration according to the updated
+  configuration and existing OSD deployment labels
+- Operator would display the status of OSD migration in the cephCluster CR status.
 
 ### API Changes
-- Add `spec.storage.store` in the Ceph cluster YAML.
+- Add `spec.storage.migration` in the CephCluster resource.
 
   ```yaml
   storage:
-    store:
-      type: bluestore-rdr
-      updateStore: yes-really-update-store
+    migration:
+        confirmation: yes-really-migrate-osds
   ```
+  - `confirmation`: Confirmation from the user that they really want to migrate the OSDs.
+    This field can only take the value `yes-really-migrate-osds`.
 
-  - `type`: The backend to be used for OSDs: `bluestore`, `bluestore-rdr`,
-     etc. The default type will be `bluestore`
-  - `updateStore`: Allows the operator to migrate existing OSDs to a different
-     backend. This field can only take the value `yes-really-update-store`. If
-     the user wants to change the `store.type` field for an existing cluster,
-     they will also need to update `spec.storage.store.updateStore` with `yes-really-update-store`.
 
+### Enable Encryption
+- Storage spec for a brownfield cluster without encryption:
+```yaml
+storage:
+   storageClassDeviceSets:
+     - name: set1
+       count: 3
+       encrypted: false
+```
+- User can enable encryption by updating the spec:
+``` yaml
+ storage:
+   migration:
+    confirmation: "yes-really-migrate-osds"
+   storageClassDeviceSets:
+     - name: set1
```

Comment:
what will happen in case of multiple device set,

What will the base value `true` or `false`

In case one device class has encryption and one dont have

Reply:
We should be able to support the scenario where only some deviceClasses have encryption enabled and some don't. It would required querying the OSDs based on the deviceClass.